packages/driver/cypress/integration/e2e/multi-domain/multi_domain_uncaught_errors_spec.ts

```diff
@@ -240,21 +257,220 @@ describe('multi-domain - uncaught errors', { experimentalSessionSupport: true },
     })
   })
 
-  it('does not fail if thrown custom error has a readonly name', (done) => {
-    cy.once('fail', (err) => {
-      expect(err.name).to.include('CustomError')
-      expect(err.message).to.include('custom error')
+  describe('unserializable errors', () => {
+    it('handles users throwing dom elements', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw document.createElement('h1')
+      })
+    })
+
+    it('handles users throwing functions', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw () => undefined
+      })
+    })
+
+    it('handles users throwing symbols', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
+
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw Symbol('foo')
+      })
+    })
+
+    it('handles users throwing promises', (done) => {
+      cy.on('fail', (err) => {
+        expect(err.name).to.equal('CypressError')
+        expect(err.message).to.equal('`cy.switchToDomain()` could not serialize or map the thrown value. Please make sure the value being thrown is supported by the structured clone algorithm.')
+        done()
+      })
 
-      done()
+      // @ts-ignore
+      cy.switchToDomain('foobar.com', () => {
+        throw new Promise(() => {})
+      })
     })
+  })
+
+  describe('serializable errors', () => {
+    it('handles users throwing complex errors/classes', (done) => {
+      cy.on('fail', (err: any) => {
+        expect(err.name).to.equal('CustomError')
+        expect(err.message).to.equal('custom error')
+        expect(err._name).to.equal('CustomError')
+        expect(err.foo).to.equal('bar')
 
-    cy.switchToDomain('foobar.com', () => {
-      cy.then(() => {
-        throw new class CustomError extends Error {
+        const { writable } = Object.getOwnPropertyDescriptor(err, 'name') || {}
```

Comment:
```suggestion
        const { writable } = Object.getOwnPropertyDescriptor(err, 'name') as PropertyDescriptor
```

Reply:
addressed in 4872216566f6ad7b26933bb99def80295f04e236